pilot/pkg/features/pilot.go

```diff
@@ -386,4 +386,18 @@ var (
 
 	WorkloadEntryHealthChecks = env.RegisterBoolVar("PILOT_ENABLE_WORKLOAD_ENTRY_HEALTHCHECKS", false,
 		"Enables automatic health checks of WorkloadEntries based on the config provided in the associated WorkloadGroup").Get()
+
+	EnableFlowControl = env.RegisterBoolVar(
+		"PILOT_ENABLE_FLOW_CONTROL",
+		false,
+		"If enabled, pilot will wait for the completion of a receive operation before"+
+			"executing a push operation. This is a form of flow control and is useful in"+
+			"environments with high rates of push requests to each gateway. By default,"+
+			"this is false.").Get()
+
+	FlowControlTimeout = env.RegisterDurationVar(
+		"PILOT_FLOW_CONTROL_TIMEOUT",
+		15*time.Second,
```

Comment:
Is this to prevent a dead lock

Reply:
I added this as a fallback incase there is a bug that causes dead lock. In theory its not needed

Reply:
good to leave it there, network error is almost unknown

Reply:
What do we do in case of timeout ?

Reply:
currently we increment a metric, log it, and then push anyways. Impact is that we will overload envoy slightly (eg as we do today), so its really not that bad to hit the timeout. Envoy overload causes slower XDS updates only, shouldn't impact data plane